src/lib/gssapi/spnego/gssapiP_spnego.h

```diff
@@ -49,6 +49,11 @@ extern "C" {
 #define	ERR_SPNEGO_NEGOTIATION_FAILED		0x20000004
 #define	ERR_SPNEGO_NO_TOKEN_FROM_ACCEPTOR	0x20000005
 
+
+#define SPNEGOINT_CHK_EMPTY(p) ((p != NULL) \
```

Comment:
Please correct my understanding if it is wrong :)

Here is my set of assumptions:

1. The mechglue only knows which mech to call when the application calls init/accept_sec_context().
2. The mechglue stores the flags when the application calls create_sec_context() and set_context_flags() due to (1)
3. If the mech determined by the mechglue is SPNEGO, the mechglue doesn't know what mech is under SPNEGO 
4. SPNEGO, once called, has no way to contact the mechglue that called it (e.g., it has no handle to the union context)
5. In its current form, when the mechglue calls SPNEGO->create_sec_context and SPNEGO->set_context_flags(), this is before the call to SPNEGO->init/accept_sec_context()
6. SPNEGO doesn't know the mech under SPNEGO->init/accept_sec_context() is called, due to creds and other features that influence its mech selection process (which might later include mech attributes due to the CB flag). 

The problem I'm trying to solve with this spnego code is that of storing the flags until (6) happens under the order in (5). Unless (4) is invalid, and we can actually access fields on the parent union_ctx, SPNEGO thus needs somewhere to store flags. To me, this was mimicking the mechglue, creating an empty SPNEGO ctx, and setting the flags there.

If you have other ideas, I'd be interested in hearing them. :)